pkg/gatewayserver/gatewayserver_test.go

```diff
@@ -122,1083 +122,1124 @@ func TestGatewayServer(t *testing.T) {
 	time.Sleep(timeout) // Wait for tasks to be started.
 
 	for _, publicLocation := range []bool{true, false} {
-		is.add(ctx, ttnpb.GatewayIdentifiers{
-			GatewayID: registeredGatewayID,
-			EUI:       &registeredGatewayEUI,
-		}, registeredGatewayKey, publicLocation)
+		for _, updateLocationFromStatus := range []bool{true, false} {
```

Comment:
Tests: change 1

Reply:
This escalates.

Can you instead write test cases with flags, like

```go
for _, := range []struct { Flag1, Flag2 bool } { { ... }, { ... } } {
   ...
}
```

Reply:
For location and updating location from status, and probably other things in the future.

We should also start planning to refactor this testing, and have the frontends test theirselves against a test bench, instead of putting the frontends as test cases, but that's not urgent.

Reply:
Yeah, it actually feels very bad in the current state. I have opened an issue for updating this already https://github.com/TheThingsNetwork/lorawan-stack/issues/1981

Not too sure if I understand the proposed change, will try and ask if something goes bad

Reply:
Tests have been re-factored to a new test case.